lib/src/layer/polyline_layer.dart

```diff
@@ -39,66 +69,191 @@ class Polyline {
+  /// possible to listen to the notifier directly.
+  ///
+  /// A [Polyline.hitKey] may be used to attach additional retrievable
+  /// information, or to signify whether a [Polyline] is "tappable", for example.
+  ///
+  /// Note that a hover event is included as a hit event. Therefore for
+  /// performance reasons, it may be advantageous to check the new value's
+  /// equality against the previous value (excluding the [PolylineHit.point],
+  /// which will always change), and avoid doing heavy work if they are the same.
+  ///
+  /// See online documentation for more detailed usage instructions. See the
+  /// example project for an example implementation.
+  ///
+  /// Will notify [PolylineHit]s if any [Polyline]s were hit, otherwise will
+  /// notify `null`.
+  final PolylineHitNotifier<TapKeyType>? hitNotifier;
+
+  /// The number of pixels away from a visual line (including any width and
+  /// outline) in which a hit should still register as a hit on the line, and
+  /// trigger a notification to [hitNotifier]
+  ///
+  /// Defaults to 3.
+  final double hitTolerance;
 
   const PolylineLayer({
     super.key,
     required this.polylines,
-    this.polylineCulling = false,
+    this.hitNotifier,
+    this.hitTolerance = 3,
+    // TODO: Remove once PR #1704 is merged
+    bool polylineCulling = true,
   });
 
   @override
   Widget build(BuildContext context) {
-    final map = MapCamera.of(context);
+    final camera = MapCamera.of(context);
 
     return MobileLayerTransformer(
       child: CustomPaint(
-        painter: PolylinePainter(
-          polylineCulling
-              ? polylines
-                  .where((p) => p.boundingBox.isOverlapping(map.visibleBounds))
-                  .toList()
-              : polylines,
-          map,
+        painter: _PolylinePainter<TapKeyType>(
+          polylines: polylines
+              .where((p) => p.boundingBox.isOverlapping(camera.visibleBounds))
+              .toList(),
+          camera: camera,
+          hitNotifier: hitNotifier,
+          onTapTolerance: hitTolerance,
         ),
-        size: Size(map.size.x, map.size.y),
+        size: Size(camera.size.x, camera.size.y),
         isComplex: true,
       ),
     );
   }
 }
 
-class PolylinePainter extends CustomPainter {
-  final List<Polyline> polylines;
-
-  final MapCamera map;
+class _PolylinePainter<TapKeyType extends Object> extends CustomPainter {
+  final List<Polyline<TapKeyType>> polylines;
+  final MapCamera camera;
   final LatLngBounds bounds;
+  final PolylineHitNotifier? hitNotifier;
+  final double onTapTolerance;
 
-  PolylinePainter(this.polylines, this.map) : bounds = map.visibleBounds;
+  _PolylinePainter({
+    required this.polylines,
+    required this.camera,
+    required this.hitNotifier,
+    required this.onTapTolerance,
+  }) : bounds = camera.visibleBounds;
 
-  int get hash => _hash ??= Object.hashAll(polylines);
+  final hits = List<Polyline<TapKeyType>>.empty(growable: true);
```

Comment:
Nice optimization! Maybe add a short comment that this is to reduce allocations

Reply:
This was your suggestion :D Resolved.